marker/src/main/java/App.java

```diff
@@ -46,19 +46,20 @@
    * @param args command line args
    */
   public static void main(String[] args) {
-
     final Logger logger = LoggerFactory.getLogger(App.class);
-    Guard guard = new Guard();
-    Thief thief = new Thief();
+    var guard = new Guard();
+    var thief = new Thief();
 
+    //noinspection ConstantConditions
     if (guard instanceof Permission) {
       guard.enter();
     } else {
       logger.info("You have no permission to enter, please leave this area");
     }
 
+    //noinspection ConstantConditions
     if (thief instanceof Permission) {
-      thief.steal();
+      thief.doNothing();
```

Comment:
With this change the if and else branches do the same thing. I would leave it to `thief.steal()` as it was.

Reply:
Pushed the changes